crypto/txscript/src/lib.rs

```diff
@@ -207,6 +361,8 @@ impl<'a, T: VerifiableTransaction, Reused: SigHashReusedValues> TxScriptEngine<'
             cond_stack: Default::default(),
             num_ops: 0,
             kip10_enabled,
+            runtime_sig_op_counter: runtime_sig_op_counting
+                .then_some(RuntimeSigOpCounter { sig_op_limit: input.sig_op_count, sig_op_remaining: input.sig_op_count }),
```

Comment:
we should use `RuntimeSigOpCounter::new` to avoid the manual double mention of `input.sig_op_count` 
(nit: I'd say perhaps this is a reason to move `RuntimeSigOpCounter ` to an inner `mod` to avoid access to the private fields)